pkg/storagev2/internal/uploadstore.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package internal
```

Comment:
So I am not sure about these. For the pinning store I have created a separate package in internal. Mostly because I think these things are completely independent. So it would be better to have it in their own package with their own test logic etc.

We could have one internal package to house the common interfaces required for these components. Like for eg the `Storage` interface that we have defined could be here.